service/create/security_group.go

```diff
@@ -55,6 +69,68 @@ func (s *Service) deleteSecurityGroup(input securityGroupInput) error {
 	return nil
 }
 
+// masterRules returns the rules for the masters security group.
+func masterRules(input rulesInput) awsresources.Rules {
+	return awsresources.Rules{
+		{
+			Port:       input.Cluster.Spec.Cluster.Kubernetes.API.SecurePort,
+			SourceCIDR: defaultCIDR,
+		},
+		{
+			Port:            input.Cluster.Spec.Cluster.Etcd.Port,
+			SecurityGroupID: input.OwnSecurityGroupID,
+		},
+		{
+			Port:       sshPort,
+			SourceCIDR: defaultCIDR,
+		},
+	}
+}
+
+// workerRules returns the rules for the workers security group.
+func workerRules(input rulesInput) awsresources.Rules {
+	return awsresources.Rules{
+		{
+			Port:            input.Cluster.Spec.Cluster.Kubernetes.IngressController.InsecurePort,
+			SecurityGroupID: input.IngressSecurityGroupID,
+		},
+		{
+			Port:            input.Cluster.Spec.Cluster.Kubernetes.IngressController.SecurePort,
+			SecurityGroupID: input.IngressSecurityGroupID,
+		},
+		{
+			Port:            input.Cluster.Spec.Cluster.Kubernetes.Kubelet.Port,
+			SecurityGroupID: input.MastersSecurityGroupID,
+		},
+		{
+			Port:       sshPort,
+			SourceCIDR: defaultCIDR,
+		},
+		{
+			Port:            calicoBGPNetworkPort,
+			SecurityGroupID: input.OwnSecurityGroupID,
+		},
+	}
```

Comment:
I like that very much. I am only wondering why these methods are not implemented on the type they receive. The definition would be better like this IMO. 

```
func (ri rulesInput) workerRules() awsresources.Rules
```